### PR TITLE
GEODE-3063 partition resolver doc revisions

### PR DESCRIPTION
**Pull request by karensmolermiller**
This section on custom partition resolvers has been substantially revised.  Please review carefully to make sure it is technically correct.

### CODE REVIEW

File: geode-docs/developing/partitioned_regions/custom_partitioning_and_data_colocation.html.md.erb [hunk 1]
```diff
@@ -31,24 +31,28 @@ This figure shows a region with customer data that is grouped into buckets by cu
...
-    -   With fixed partitioning, the region configuration is different between host members. Each member identifies the named partitions it hosts, and whether it is hosting the primary copy or a secondary copy. You then program fixed partition resolver to return the partition id, so the entry is placed on the right members. Only one member can be primary for a particular partition name and that member cannot be the partition's secondary.
+    -   <%=vars.product_name%> cannot rebalance fixed partition region data, because it cannot move the buckets around among the host members. You must carefully consider your expected data loads for the partitions you create.
+    -   With fixed partitioning, the region configuration is different between host members. Each member identifies the named partitions it hosts, and whether it is hosting the primary copy or a secondary copy. You then program fixed partition resolver to return the partition id, so the entry is placed on the right members. Only one member can be primary for a particular partition name, and that member cannot be the partition's secondary.
```
Comment (joeymcallister) [inline geode-docs/developing/partitioned_regions/custom_partitioning_and_data_colocation.html.md.erb]: Add "a" and hyphenate "program a fixed-partition resolver"

File: geode-docs/developing/partitioned_regions/custom_partitioning_and_data_colocation.html.md.erb [hunk 2]
```diff
@@ -31,24 +31,28 @@ This figure shows a region with customer data that is grouped into buckets by cu
...
 
     Fixed partitioning has these requirements and caveats:
 
-    -   <%=vars.product_name%> cannot rebalance fixed partition region data because it cannot move the buckets around among the host members. You must carefully consider your expected data loads for the partitions you create.
-    -   With fixed partitioning, the region configuration is different between host members. Each member identifies the named partitions it hosts, and whether it is hosting the primary copy or a secondary copy. You then program fixed partition resolver to return the partition id, so the entry is placed on the right members. Only one member can be primary for a particular partition name and that member cannot be the partition's secondary.
+    -   <%=vars.product_name%> cannot rebalance fixed partition region data, because it cannot move the buckets around among the host members. You must carefully consider your expected data loads for the partitions you create.
```
Comment (joeymcallister) [inline geode-docs/developing/partitioned_regions/custom_partitioning_and_data_colocation.html.md.erb]: hyphenate "fixed-partition region data"

File: geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb [hunk 3]
```diff
@@ -19,47 +19,63 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-By default, <%=vars.product_name%> partitions each data entry into a bucket using a hashing policy on the key. Additionally, the physical location of the key-value pair is abstracted away from the application. You can change these policies for a partitioned region. You can provide your own data partitioning resolver and you can additionally specify which members host which data buckets.
+By default, <%=vars.product_name%> partitions each data entry into a bucket using a hashing policy on the key. 
+Additionally, the physical location of the key-value pair 
+is abstracted away from the application.
+You can change these policies for a partitioned region by providing
+your own partition resolver.
+The partitioning can go farther with a fixed partition resolver
```
Comment (joeymcallister) [inline geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb]: Change "farther" to "further"

Comment (joeymcallister) [reply geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb]: hyphenate "fixed-partition resolver"

File: geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb [hunk 4]
```diff
@@ -226,15 +246,25 @@ program the resolver's `getRoutingObject` method to return the routing object fo
     -   **Entry key**. Use the key object with the resolver implementation for every entry operation.
     -   **Cache callback argument**. Provide the argument to every call that accesses an entry. This restricts you to calls that take a callback argument.
 
-    If using the default implementation of the string-based partition resolver, configure with one of:
+-   If your colocated data is in a server system,
+add the `PartitionResolver` implementation class to the `CLASSPATH`
+of your Java clients. 
+For Java single hop access to work,
```
Comment (joeymcallister) [inline geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb]: hyphenate "single-hop access"

File: geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb [hunk 5]
```diff
@@ -243,21 +273,85 @@ program the resolver's `getRoutingObject` method to return the routing object fo
...
     ```
     **gfsh:**
-
+    
     Add the option `--partition-resolver=org.apache.geode.cache.util.StringPrefixPartitionResolver` to the `gfsh create region` command.
 
-6.  If your colocated data is in a server system, add the `PartitionResolver` implementation class to the `CLASSPATH` of your Java clients. The resolver is used for single hop access to partitioned region data in the servers.
+-  For colocated data, add the `StringPrefixPartitionResolver`
+implementation class to the `CLASSPATH` of your Java clients.
+The resolver will work with Java single hop clients.
```
Comment (joeymcallister) [inline geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb]: hyphenate "single-hop clients"

File: geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb [hunk 6]
```diff
@@ -243,21 +273,85 @@ program the resolver's `getRoutingObject` method to return the routing object fo
...
+
+**Configuring Fixed Partitioning**
+
+-  Set the fixed partition attributes for each member.
```
Comment (joeymcallister) [inline geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb]: hyphenate "fixed-partition attributes"

File: geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb [hunk 7]
```diff
@@ -243,21 +273,85 @@ program the resolver's `getRoutingObject` method to return the routing object fo
...
+
+    These attributes define the data stored for the region by the member and must be different for different members. See `org.apache.geode.cache.FixedPartitionAttributes` for definitions of the attributes. Define each `partition-name` in your data host members for the region. For each partition name, in the member you want to host the primary copy, define it with `is-primary` set to `true`. In every member you want to host the secondary copy, define it with `is-primary` set to `false` (the default). The number of secondaries must match the number of redundant copies you have defined for the region. See [Configure High Availability for a Partitioned Region](configuring_ha_for_pr.html).
```
Comment (joeymcallister) [inline geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb]: hyphenate "data-host members"

File: geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb [hunk 8]
```diff
@@ -243,21 +273,85 @@ program the resolver's `getRoutingObject` method to return the routing object fo
...
+            .create("Trades");
+        ```
+    -   gfsh:
+
+        You cannot specify a fixed partition resolver using gfsh.
+
+-   If your colocated data is in a server system,
+add the class that implements the `FixedPartitionResolver` interface
+to the `CLASSPATH` of your Java clients.
+For Java single hop access to work,
```
Comment (joeymcallister) [inline geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb]: hyphenate "single-hop access"

File: geode-docs/developing/partitioned_regions/colocating_partitioned_region_data.html.md.erb [hunk 9]
```diff
@@ -19,27 +19,19 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-By default, <%=vars.product_name%> allocates the data locations for a partitioned region independent of the data locations for any other partitioned region. You can change this policy for any group of partitioned regions, so that cross-region, related data is all hosted by the same member. This colocation speeds queries and other operations that access data from the regions.
+By default, <%=vars.product_name%> allocates the data locations for a partitioned region independent of the data locations for any other partitioned region. You can change this policy for any group of partitioned regions, so that cross-region, related data is all hosted by the same member.
+Colocation is required for some operations,
```
Comment (davebarnes97) [inline geode-docs/developing/partitioned_regions/colocating_partitioned_region_data.html.md.erb]: This paragraph also appears in overview_custom_partitioning_and_data_colocation.html, so you need to repeat this edit in that file, too.

File: geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb [hunk 10]
```diff
@@ -19,47 +19,64 @@ See the License for the specific language governing permissions and
...
 **Note:**
-If you are colocating data between regions and custom partitioning the data in the regions, all colocated regions must use the same custom partitioning mechanism. See [Colocate Data from Different Partitioned Regions](colocating_partitioned_region_data.html#colocating_partitioned_region_data).
+If you are both colocating region data and custom partitioning,
+all colocated regions must use the same custom partitioning mechanism.
+See [Colocate Data from Different Partitioned Regions](colocating_partitioned_region_data.html#colocating_partitioned_region_data).
 
-<a id="custom_partition_region_data__section_1D7043815DF24308ABE4C78BFDFEE686"></a>
+The two steps of implementing the interface
+and configuring the regions are described.
+These steps differ based on which partition resolver is used.
```
Comment (davebarnes97) [inline geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb]: Consider active voice here, maybe something like:

To custom-partition your region data, follow two steps:
- implement the interface
- configure the regions
These steps differ based on which partition resolver you use.

Comment (karensmolermiller) [reply geode-docs/developing/partitioned_regions/using_custom_partition_resolvers.html.md.erb]: Thanks Dave!  I made these changes.

**Pull request merged**